### PR TITLE
Fix empty screen issue after clearing the cache

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Ensure that after a clear cache the application are still usable.

## Motivation and context

Avoid dead lock.

## Tests

- From the developer setting, clear the cache
- Before: blank screen
- Now: analytics opt in screen is displayed.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/state/DefaultFtueService.kt [hunk 1]
```diff
@@ -119,18 +122,14 @@ class DefaultFtueService @Inject constructor(
                 emit(SessionVerifiedStatus.NotVerified)
             }
             .first()
-        // For some obscure reason we need to call this *before* we check the `readyVerifiedSessionStatus`, otherwise there's a deadlock
-        // It seems like a DataStore bug
-        val skipVerification = canSkipVerification()
-        return readyVerifiedSessionStatus == SessionVerifiedStatus.NotVerified && !skipVerification
+        return readyVerifiedSessionStatus == SessionVerifiedStatus.NotVerified && !canSkipVerification()
```
Comment (bmarty) [inline features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/state/DefaultFtueService.kt:125]: It seems that the previous workaround is not needed anymore CC @jmartinesp

Comment (jmartinesp) [reply features/ftue/impl/src/main/kotlin/io/element/android/features/ftue/impl/state/DefaultFtueService.kt:125]: Are you sure? Did you try logging out and then in again, and starting the app with no network connection? I'm surprised this issue fixed itself.

File: appnav/src/main/kotlin/io/element/android/appnav/root/RootNavStateFlowFactory.kt [hunk 2]
```diff
@@ -73,6 +75,8 @@ class RootNavStateFlowFactory @Inject constructor(
                 matrixClientsHolder.remove(sessionId)
                 // Ensure image loader will be recreated with the new MatrixClient
                 imageLoaderHolder.remove(sessionId)
+                // Also remove cached value for SessionPreferencesStore
+                sessionPreferencesStoreFactory.remove(sessionId)
```
Comment (bmarty) [inline appnav/src/main/kotlin/io/element/android/appnav/root/RootNavStateFlowFactory.kt:79]: This is actually the location of the fix for the blank screen after clear cache.

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**